### PR TITLE
feat(inventory in stock) display exclusively user-accessible depots' lots

### LINKED ISSUE #2637: Search for depots limits to only user-accessible depots
When I want to find stock in a depot, I shouldn't require permission to use the depot.  Interestingly, the items from a depot are displayed in the stock list even if the user doesn't have access to that depot, but they are limited from looking up that depot.  See below:

_Fig 1: Cannot find depot "Project Nutrition"_

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #2637

### CODE REVIEW

File: server/controllers/stock/core.js [hunk 1]
```diff
@@ -98,6 +99,7 @@ function getLots(sqlQuery, parameters, finalClauseParameter) {
   filters.equals('group_uuid', 'uuid', 'ig');
   filters.equals('text', 'text', 'i');
   filters.equals('label', 'label', 'l');
+  filters.custom('user_id', 'dp.user_id=?');
```
Comment (mbayopanda) [inline server/controllers/stock/core.js:102]: 👍

File: server/controllers/stock/core.js [hunk 2]
```diff
@@ -209,7 +211,7 @@ function getLotsMovements(depotUuid, params) {
 
   const sql = `
         SELECT
-          BUID(l.uuid) AS uuid, l.label, l.initial_quantity, m.quantity, m.reference, m.description,
+          DISTINCT BUID(l.uuid) AS uuid, l.label, l.initial_quantity, m.quantity, m.reference, m.description,
```
Comment (mbayopanda) [inline server/controllers/stock/core.js]: Could you remove this DISTINCT, it is not necessary in the context of this query

**Review by mbayopanda — COMMENTED** [review]
@jeremielodi could you update this PR according comments in this review.

- The DISTINCT is still there

**Comment by mbayopanda** [comment]
@jeremielodi could you update this PR ?

**Comment by jniles** [comment]
bors r=mbayopanda

**Pull request merged**